ores/score_request.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+
+class ScoreRequest:
+    def __init__(self, context_name, rev_ids, model_names, precache=False,
+                 include_features=False, injection_caches=None,
+                 model_info=None):
+        self.context_name = context_name
+        self.rev_ids = set(rev_ids)
+        self.model_names = set(model_names)
+        self.precache = precache
+        self.include_features = include_features
+        self.injection_caches = injection_caches or {}
+        self.model_info = model_info
+
+    def __str__(self):
+        return self.format()
+
+    def format(self, rev_id=None, model_name=None):
```

Comment:
This and __repr__ below would benefit from some docstrings.  What's the difference?  Why would I want to cast to string and how would I use the result?  If it's a magic format, maybe there should be more semantic accessors like "get_unique_key" or sumsuch.